src/extractor/extraction_containers.cpp

```diff
@@ -625,6 +635,40 @@ void ExtractionContainers::WriteNodes(storage::io::FileWriter &file_out) const
         log << "ok, after " << TIMER_SEC(write_nodes) << "s";
     }
 
+    {
+        util::UnbufferedLog log;
+        log << "Writing barrier nodes     ... ";
+        TIMER_START(write_nodes);
+        std::vector<NodeID> internal_barrier_nodes;
+        for (const auto id : barrier_nodes)
+        {
+            auto iter = external_to_internal_node_id_map.find(id);
+            if (iter != external_to_internal_node_id_map.end())
```

Comment:
Yes. `external_to_internal_node_id_map` only has entries for nodes that we actually use in the road network. However we do parse all nodes.